optimum/habana/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_instruct_pix2pix.py

```diff
@@ -0,0 +1,590 @@
+            )
+
+            height, width = image_latents.shape[-2:]
+            height = height * self.vae_scale_factor
+            width = width * self.vae_scale_factor
+
+            # 6. Prepare latent variables
+            num_channels_latents = self.vae.config.latent_channels
+            latents = self.prepare_latents(
+                num_prompts * num_images_per_prompt,
+                num_channels_latents,
+                height,
+                width,
+                prompt_embeds.dtype,
+                device,
+                generator,
+                latents,
+            )
+            # 7. Check that shapes of latents and image match the UNet channels
+            num_channels_image = image_latents.shape[1]
+            if num_channels_latents + num_channels_image != self.unet.config.in_channels:
+                raise ValueError(
+                    f"Incorrect configuration settings! The config of `pipeline.unet`: {self.unet.config} expects"
+                    f" {self.unet.config.in_channels} but received `num_channels_latents`: {num_channels_latents} +"
+                    f" `num_channels_image`: {num_channels_image} "
+                    f" = {num_channels_latents+num_channels_image}. Please verify the config of"
+                    " `pipeline.unet` or your `image` input."
+                )
+
+            # 8. Prepare extra step kwargs. TODO: Logic should ideally just be moved out of the pipeline
+            extra_step_kwargs = self.prepare_extra_step_kwargs(generator, eta)
+
+            # 8.1 Add image embeds for IP-Adapter
+            added_cond_kwargs = {"image_embeds": image_embeds} if ip_adapter_image is not None else None
+
+            # 9. Split into batches (HPU-specific step)
+            latents_batches, prompt_embeds_batches, image_latents_batches, num_dummy_samples = (
+                self._split_inputs_into_batches(
+                    batch_size,
+                    latents,
+                    prompt_embeds,
+                    image_latents,
+                    self.do_classifier_free_guidance,
+                )
+            )
+            outputs = {
+                "images": [],
+                "has_nsfw_concept": [],
+            }
+            hb_profiler = HabanaProfile(
+                warmup=profiling_warmup_steps,
+                active=profiling_steps,
+                record_shapes=False,
+            )
+            hb_profiler.start()
+
+            # 10. Denoising loop
+            t0 = time.time()
+            t1 = t0
+            throughput_warmup_steps = kwargs.get("throughput_warmup_steps", 3)
+            num_warmup_steps = len(timesteps) - num_inference_steps * self.scheduler.order
+            for j in self.progress_bar(range(num_batches)):
+                # The throughput is calculated from the 3rd iteration
+                # because compilation occurs in the first two iterations
+                if j == throughput_warmup_steps:
```

Comment:
Please use updated logic for warmup here. See https://github.com/huggingface/optimum-habana/blob/main/optimum/habana/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L498
https://github.com/huggingface/optimum-habana/blob/main/optimum/habana/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L507
https://github.com/huggingface/optimum-habana/blob/main/optimum/habana/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L601